apps/backend/src/auth/auth.middleware.ts

```diff
@@ -1,35 +1,32 @@
 import Elysia from "elysia";
 import { jwt } from "../plugins";
 import { prisma } from "@nutrishare/db";
+import { UnauthorizedError } from "../errors";
 
-class UnauthorizedError extends Error {}
-
-const authMiddleware = new Elysia()
+export default new Elysia()
   .use(jwt)
-  .addError({
+  .error({
     UnauthorizedError,
   })
-  .derive(async ({ jwt, headers, set }) => {
+  .derive(async ({ jwt, headers }) => {
     // biome-ignore lint: Valid Record access
     const accessToken = headers["authorization"]?.split(" ")[1];
-    if (!accessToken) {
-      set.status = 401;
-      throw new UnauthorizedError();
-    }
+    if (!accessToken) throw new UnauthorizedError();
 
-    const result = await jwt.verify(accessToken);
-    if (!result) {
-      set.status = 401;
-      throw new UnauthorizedError();
-    }
+    const jwtPayload = await jwt.verify(accessToken);
+    if (!jwtPayload) throw new UnauthorizedError();
 
-    const { id } = result;
+    const { id } = jwtPayload;
     const user = await prisma.user.findFirst({ where: { id } });
-    if (!user) {
-      set.status = 401;
-      throw new UnauthorizedError();
-    }
+    if (!user) throw new UnauthorizedError();
 
     return { user };
+  })
+  .onError(({ code, error, set }) => {
+    if (code === "UnauthorizedError") {
+      set.status = "Unauthorized";
+      set.headers["WWW-Authenticate"] = "Bearer";
+      return error.message;
```

Comment:
I'm unsure about this handling, how does it look like if we hit it?

Reply:
It returns a 403 error, and the error's message is by default "Unauthorized".

Reply:
I'm not sure about how `WWW-Authenticate` should be used. We should research it more before we use it in every unauthorized request response. 😺